native/packages/react-native-bpk-theming/src/BpkThemeProvider.js

```diff
@@ -16,11 +16,70 @@
  * limitations under the License.
  */
 
- import { ThemeProvider, withTheme } from 'theming';
+ import React from 'react';
+ import PropTypes from 'prop-types';
+ import { ThemeProvider } from 'theming';
+
+ // This is the same as https://github.com/iamstarkov/theming/blob/master/src/create-theme-listener.js
+ // but returns null instead of an error when context[CHANNEL] is undefined.
+ const createThemeListener = () => {
```

Comment:
Can we split these functions into separate files, similar to the upstream example?